### DISCUSSION TITLE
Rate limiting configuration

### DISCUSSION THREAD

**Discussion by hholstil** [emqx/emqx#13751] [Q&A]
I would like to configure EMQX to rate limit messages from clients but to allow some burst capacity.
It looks like in the past there has been a way to configure it, but nowadays the only documented configuration value is `listener.messages_rate`.

It looks like this could be achieved with the undocumented/deprecated configuration, for example:
`
limiter {
    client {
      messages {
        rate = "2/s"
        burst = "8/s"
        initial = 8
      }
    }
  }
`

So my questions are
- What is the reason this more flexible way of configuring the rate limiting has been deprecated/undocumented?
- Is there a plan to replace it with something in the future?
- Is there a plan to remove it completely in the future on can I keep using it regardless of it being undocumented?

**Comment by lafirest** [comment]
@hholstil 
Sorry for that, since the flexible way was blamed as too complex, unlike similar products that only need to configure one field, the flexible way was deprecated.
Currently, the flexible way is only kept for compatibility,  and there is no plan to enable it again.

**Comment by 04116** [comment]
Hi, maybe for other topic. But, if I use mqtt at server side. How to config it not effected by the ratelimit?
